packages/trader/src/App/Components/Elements/ContractAudit/contract-details.jsx

```diff
@@ -32,16 +35,22 @@ const ContractDetails = ({ contract_end_time, contract_info, duration, duration_
         entry_spot_display_value,
         entry_tick_time,
         exit_tick_time,
+        high_barrier,
         profit,
         date_start,
         tick_count,
         tick_passed,
         transaction_ids: { buy, sell } = {},
+        low_barrier,
         number_of_contracts,
     } = contract_info;
 
     const is_profit = profit >= 0;
     const cancellation_price = getCancellationPrice(contract_info);
+    const show_barrier = !is_vanilla && !isAccumulatorContract(contract_type) && !isSmartTraderContract(contract_type);
+    const show_duration = !isAccumulatorContract(contract_type) || !isNaN(contract_end_time);
+    const show_payout_per_point = is_vanilla;
+    const show_strike_barrier = is_vanilla || isAsiansContract(contract_type);
```

Comment:
Add all this constants for refactoring purposes: code was covered with different checks of type and I decided to unite and lift up most of them here.